lib/sycamore/sycamore/connectors/weaviate/weaviate_scan.py

```diff
@@ -0,0 +1,26 @@
+from dataclasses import asdict
+from sycamore.connectors.weaviate.weaviate_writer import WeaviateClientParams
+from sycamore.plan_nodes import Scan
+from sycamore.data import Document
+from weaviate import WeaviateClient
+from ray.data import Dataset, from_items
+
+
+class WeaviateScan(Scan):
+    def __init__(self, collection_name: str, connection_params: WeaviateClientParams, **kwargs):
+        super().__init__(**kwargs)
+        self._collection_name = collection_name
+        self._connection_params = connection_params
+
+    def execute(self) -> Dataset:
+        documents = []
+        with WeaviateClient(**asdict(self._connection_params)) as wcl:
+            collection = wcl.collections.get(self._collection_name)
+            for object in collection.iterator(include_vector=True):
+                doc = Document(object.vector | object.properties | {"doc_id": str(object.uuid)})  # type: ignore
```

Comment:
Wouldn't you want to leave the dictionary from `object.properties` as is and not merge its parts? To ensure it matches the input version

Reply:
object.properties is {"text_representation": blah, "type": blah, "properties": nested blah, etc}. We need to tack on the embedding vector and doc_id bc those aren't indexed as weaviate 'properties'

Reply:
Ah I see, thanks for clarifying